### PR TITLE
Switch over to rapids-logger

### PR DESCRIPTION
**Pull request by vyasr**
This PR removes raft's implementation of a logger in favor of the centralized one in [rapids-logger](https://github.com/rapidsai/rapids-logger). Consumers still get the benefits of a PImpl idiom, but now that is primarily handled by using the appropriate targets (if necessary the impl header is of course still available for direct inclusion). This change paves the way for ensuring consistent fmt/spdlog (lack of) linkage throughout RAPIDS conda and wheel packages.

This PR requires https://github.com/rapidsai/rapids-logger/pull/1

Contributes to https://github.com/rapidsai/build-planning/issues/104

### CODE REVIEW

**Comment by vyasr** [comment]
I'll be on vacation next week and this PR has downstream consequences so I've marked it as "DO NOT MERGE" until I get back, but I would love to get it reviewed in advance so that I can safely merge when I'm ready.

**Review by bdice — APPROVED** [review]

File: cpp/CMakeLists.txt
Comment (cjnolet) [inline cpp/CMakeLists.txt:169]: Why feat/cuml here?

**Review by cjnolet — COMMENTED** [review]
The changes look fine overall. Definitely good to see consolidation. Just a question (though not sure if it’s important to merge this).

**Review by cjnolet — APPROVED** [review]
Pre-approving because the changes look fine to me

File: cpp/CMakeLists.txt
Comment (vyasr) [reply cpp/CMakeLists.txt:169]: This is only because I started work on https://github.com/rapidsai/cuml/pull/6162 before this PR, so that's the branch name for https://github.com/rapidsai/rapids-logger/pull/1. I'll merge that PR and update the commit hash here before this PR is merged.

Comment (vyasr) [reply cpp/CMakeLists.txt]: Done.

**Pull request merged**
